deploy_windows.ps1

```diff
@@ -1,202 +1,208 @@
-    Remove-Item -Recurse -Force (Join-Path -Path $binDir -ChildPath $UPX_FOLDER)
-    "Removing zip file"
-    Remove-Item $TEMP_UPX_ZIP
-}
 
-# Download mimikatz binaries
-$mk32_path = Join-Path -Path $binDir -ChildPath $MK32_DLL
-if(!(Test-Path -Path $mk32_path )){
-    "Downloading mimikatz 32 binary"
-    $webClient.DownloadFile($MK32_DLL_URL, $mk32_path)
-}
-$mk64_path = Join-Path -Path $binDir -ChildPath $MK64_DLL
-if(!(Test-Path -Path $mk64_path )){
-    "Downloading mimikatz 32 binary"
-    $webClient.DownloadFile($MK64_DLL_URL, $mk64_path)
-}
+    # We check if python is installed
+    try
+    {
+        $version = cmd.exe /c '"python" --version  2>&1'
+        if ( $version -like 'Python 2.7.*' ) {
+            "Python 2.7.* was found, installing dependancies"
+        } else {
+            throw System.Management.Automation.CommandNotFoundException
+        }
+    }
+    catch [System.Management.Automation.CommandNotFoundException]
+    {
+        "Downloading python 2.7 ..."
+        $webClient.DownloadFile($PYTHON_URL, $TEMP_PYTHON_INSTALLER)
+        Start-Process -Wait $TEMP_PYTHON_INSTALLER -ErrorAction Stop
+        $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine")
+        Remove-Item $TEMP_PYTHON_INSTALLER
+        # Check if installed correctly
+        $version = cmd.exe /c '"python" --version  2>&1'
+        if ( $version -like '* is not recognized*' ) {
+            "Python is not found in PATH. Add it manually or reinstall python."
+            return
+        }
+    }
+
+    # Set python home dir
+    $PYTHON_PATH = Split-Path -Path (get-command python | Select-Object -ExpandProperty Source)
+
+    # Get vcforpython27 before installing requirements
+    "Downloading Visual C++ Compiler for Python 2.7 ..."
+    $webClient.DownloadFile($VC_FOR_PYTHON27_URL, $TEMP_VC_FOR_PYTHON27_INSTALLER)
+    Start-Process -Wait $TEMP_VC_FOR_PYTHON27_INSTALLER -ErrorAction Stop
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") 
+    Remove-Item $TEMP_VC_FOR_PYTHON27_INSTALLER
+
+    # Install requirements
+    $reqPath = Join-Path -Path $monkey_home -ChildPath $MONKEY_ISLAND_DIR | Join-Path -ChildPath "\requirements.txt" -ErrorAction Stopd
+    "Upgrading pip..."
+    $output = cmd.exe /c 'python -m pip install --upgrade pip 2>&1'
+    $output
+    if ( $output -like '*No module named pip*' ) {
+        "Make sure pip module is installed and re-run this script."
+        return
+    }
+    & python -m pip install -r $reqPath
+    # Install requirements from monkey to be able to develop monkey itself
+    $reqPath = Join-Path -Path $monkey_home -ChildPath $MONKEY_DIR | Join-Path -ChildPath "\requirements.txt"
+    & python -m pip install -r $reqPath
+
+    # Transfer python file to local directory
+    "Copying python folder to bin"
+    Copy-Item $PYTHON_PATH -Destination (Join-Path -Path $binDir -ChildPath "Python27") -Recurse -ErrorAction SilentlyContinue
```

Comment:
Why do this for this env if it's for dev. This can easily confuse people adding requirements.

Reply:
If we need this in terms of paths, I'd suggest adding a symlink and not adding a lot of stuff to the drive.